pkg/plugins/autodiscovery/cargo/dependencyManifest.go

```diff
@@ -0,0 +1,81 @@
+package cargo
+
+const (
+	// dependencyManifest is the Go template used to generate
+	// the manifests to update the cargo file
+	dependencyManifest string = `name: '{{ .ManifestName }}'
+sources:
+  {{ .SourceID }}:
+    name: '{{ .SourceName }}'
+    kind: 'cargopackage'
+    spec:
+      package: '{{ .DependencyName }}'
+      versionFilter:
+        kind: '{{ .SourceVersionFilterKind }}'
+        pattern: '{{ .SourceVersionFilterPattern }}'
+{{- if .WithRegistry }}
+      registry:
+        url: '{{ .RegistryURL }}'
+        rootdir: '{{ .RegistryRootDir }}'
+        auth:
+          token: '{{ .RegistryAuthToken }}'
+          headerFormat : '{{ .RegistryHeaderFormat }}'		
+{{- if .RegistrySCMID }}
+    scmid: '{{ .RegistrySCMID }}'
+{{- end }}
+{{- end }}
+  {{ .ExistingSourceID }}:
+    name: '{{ .ExistingSourceName }}'
+    kind: 'toml'
+{{- if .ScmID }}
+    scmid: '{{ .ScmID }}'
+{{- end }}
+    spec:
+      file: '{{ .File }}'
+      Key: '{{ .ExistingSourceKey }}'
+conditions:
+  {{ .ConditionID }}:
+    name: 'Ensure Cargo chart named "{{ .DependencyName }}" is specified'
+    kind: 'toml'
+{{- if .ScmID }}
+    scmid: '{{ .ScmID }}'
+{{- end }}
+    spec:
+      file: '{{ .File }}'
+      query: '{{ .ConditionQuery }}'
+    sourceid: '{{ .ExistingSourceID }}'
+targets:
+  {{ .TargetID }}:
+    name: 'Bump crate dependency "{{ .DependencyName }}" for crate "{{ .CrateName }}"'
+    kind: 'toml'
+{{- if .ScmID }}
+    scmid: '{{ .ScmID }}'
+{{- end }}
+    spec:
+      file: '{{ .TargetFile }}'
+      key: '{{ .TargetKey }}'
```

Comment:
An issue I faced on the npm pullrequest is the situation where I want the dependency "~4.0". The lockfile contains "4.0.0" but "4.0.10" is available. I don't want to update the manifest from "~4.0" to "4.0.10", I only update the lockfile.

I am wondering how it works with cargo

Reply:
It appears you initial approach was already good